common/Util.cpp

```diff
@@ -131,7 +138,13 @@ namespace Util
             // a poor fallback but something.
             std::vector<char> random = getBytes(length);
             int len = 0;
-#ifdef HAVE_SYS_RANDOM_H
+#ifdef BSD
+            len = getentropy(random.data(), length);
+
+            // if getentropy() fails, it will return -1.
+            // then, we should fall back to "/dev/[u]random" approach.
```

Comment:
I guess we could say `we fall back to getrandom(), if available, otherwise we try "/dev/[u]random".` (or something similar).

Reply:
> I guess we could say `we fall back to getrandom(), if available, otherwise we try "/dev/[u]random".` (or something similar).

But if we are in `#ifdef BSD` code block and  `getentropy()` fails in here, then we can't(?) execute `#elif defined HAVE_SYS_RANDOM_H` code clock - because it is `#elif` directive? Then we can't fall back to `getrandom()`, but we can only use `"/dev/[u]random"` - if `getentropy()` fails. Therefore, imho, we can(?) only use something like `// if getentropy() fails, try "/dev/[u]random"` - if we stick with the current approach.

but:

If we want to fall back to `getrandom()` if `getentropy()` fails in `#ifdef BSD` code block(as you stated), then I can update the code accordingly.

(to sum up) Which one we should use? :
1) if `BSD` -> use `getentropy()` -> if `getentropy()` fails -> (skip `getrandom()`) -> use `"/dev/urandom"`. (current approach)
or
2) if `BSD` -> use `getentropy()` -> if `getentropy()` fails -> use `getrandom()` -> if `getrandom()` fails -> use `"/dev/urandom"`.

Thanks for the review.

Reply:
According to man, `getentropy` fails either with `EFAULT` due to invalid buffer pointer or `EIO` in case of "some other fatal error", which probably means that there's something really wrong with the system.

And at least of FreeBSD `getrandom` does the same as `getentropy`, so I think there is no point in trying both functions.